src/main/java/com/artipie/scripting/Script.java

```diff
@@ -0,0 +1,226 @@
+
+        /**
+         * Ctor.
+         * @param type Name of scripting engine.
+         * @param script Script code
+         */
+        public PrecompiledScript(final ScriptType type, final String script) {
+            final ScriptEngine engine = Script.MANAGER.getEngineByName(type.toString());
+            if (!(engine instanceof Compilable)) {
+                throw new ArtipieException(
+                    String.format("Scripting engine '%s' does not support compilation", engine)
+                );
+            }
+            try {
+                this.script = ((Compilable) engine).compile(script);
+            } catch (final ScriptException exc) {
+                throw new ArtipieException(exc);
+            }
+        }
+
+        @Override
+        public Result call() throws ScriptException {
+            final Result result = new Result();
+            result.value(this.script.eval(result.context()));
+            return result;
+        }
+
+        @Override
+        public Result call(final Map<String, Object> vars) throws ScriptException {
+            final Result result = new Result(vars);
+            result.value(this.script.eval(result.context()));
+            return result;
+        }
+    }
+
+    /**
+     * Result of script invocation.
+     * @since 0.30
+     */
+    @SuppressWarnings(
+        {
+            "PMD.OnlyOneConstructorShouldDoInitialization",
+            "PMD.ConstructorOnlyInitializesOrCallOtherConstructors"
+        }
+    )
+    class Result {
+        /**
+         * Script context.
+         */
+        private final ScriptContext context;
+
+        /**
+         * Resulting value on script evaluation (like eval()).
+         */
+        private Object val;
+
+        /**
+         * Ctor.
+         */
+        public Result() {
+            this.context = new SimpleScriptContext();
+        }
+
+        /**
+         * Ctor.
+         * @param vars Environment variables
+         */
+        public Result(final Map<String, Object> vars) {
+            this();
+            this.context.setBindings(new SimpleBindings(vars), ScriptContext.ENGINE_SCOPE);
+        }
+
+        /**
+         * Resulting value on script evaluation.
+         * @return Value
+         */
+        public Object value() {
+            return this.val;
+        }
+
+        /**
+         * Environment variable by name.
+         * @param name Environment variable
+         * @return Environment variable or null, if not found.
+         */
+        public Object variable(final String name) {
+            return this.context.getBindings(ScriptContext.ENGINE_SCOPE).get(name);
+        }
+
+        /**
+         * Script context.
+         * @return ScriptContext
+         */
+        private ScriptContext context() {
+            return this.context;
+        }
+
+        /**
+         * Setter for resulting value.
+         * @param value Resulting value
+         */
+        private void value(final Object value) {
```

Comment:
I still do not get why the `val` field, **private** setter and **public** getter are required... Setter is not called from anywhere in this class...

Reply:
@olenagerasimova , That setter is called from `com.artipie.scripting.Script.PrecompiledScript#call()`. I renamed it to `setValue()` and added basic test for `eval()` on expression, which also demonstrates usage.